lib/nebulex_redis_adapter/redis_cluster/config_manager.ex

```diff
@@ -302,6 +285,48 @@ defmodule NebulexRedisAdapter.RedisCluster.ConfigManager do
     end)
   end
 
+  defp extract_endpoint_and_port([
```

Comment:
I was thinking, perhaps we can re-write this function like this:

```elixir
defp parse_node_attrs(nodes) do
  Enum.reduce(nodes, [], fn attrs, acc ->
    attrs =
      attrs
      |> Enum.chunk_every(2)
      |> Enum.reduce(%{}, fn [key, value], acc ->
        Map.put(acc, key, value)
      end)

    [attrs | acc]
  end)
  |> Enum.filter(&(&1["role"] == "master" and &1["health"] == "online"))
end
```

And from the calling part:

```elixir
case parse_node_attrs(nodes) do
  [] ->
    # coveralls-ignore-start
    acc

  # coveralls-ignore-stop

  [attrs | _] ->
    host = attrs["endpoint"]
    port = attrs["tls-port"] || attrs["port"]

    [{start, stop, host, port} | acc]
end
```

That will also help with the test part. Please try it out and let me know WDYT.

Reply:
Thanks @cabol I think that works better! I will rewrite it as you suggested.

Reply:
@cabol I'm trying to test mocking the `Redix.command(conn, ["CLUSTER", "SHARDS"])` to return the expected values, but it is not working. What do you think? Is this the way to go?

```elixir
      expect(Redix, :command, fn _, ["CLUSTER", "SHARDS"] ->
        {:ok,
         [
           [
             "slots",
             [0, 16383],
             "nodes",
             [node1, node2, ...]
         ]}
      end)

      {:ok, pid} = Cache.start_link(name: :redis_cluster_optional_fields)
 ```

```
** (Mimic.VerificationError) error while verifying mocks for #PID<0.542.0>:
     
       * expected Redix.command/2 to be invoked 1 time(s) but it has been called 0 time(s)
```

Reply:
Hey @norbajunior !!

The issue is the `expect` is applied to the process given by `self()` and the command on the other hand is executed by another process, part of the cache supervision tree. This may be a bit tricky to test, I would skip adding a test at the moment, I think it is okay because it is covered anyway. The only uncovered part is when TLS is used, but we know it works because we are using the right keys to get the port (`"tls-port"`). So, no worries, just add the fix without tests, the coverage will work anyways.

Reply:
Ok then, thank you!